snaps/utils.go

```diff
@@ -138,3 +140,10 @@ func shouldCreate(u *bool) bool {
 
 	return true
 }
+
+// trimPathBuild checks if the build has trimpath setting true
+func trimPathBuild() bool {
+	goFlags := strings.Split(os.Getenv("GOFLAGS"), " ")
+
+	return slices.Contains(goFlags, "-trimpath")
```

Comment:
Golang accepts both `-trimpath` and `--trimpath` - and I know that due to old habits I tend to use `--` interactively. Maybe it would be worth checking for both.

Reply:
ohhhh thanks for pointing this out 🙇 . Btw not sure but I can't find a "good" way to identify if build is trimpath when you run your tests like this 

```
go test -trimpath ./...
```

if you have any ideas feel free to share them 😄

Reply:
I found this but unfortunately this information isn't always captured. 

The previous commit on this pr was using the `ReadBuildInfo`, but when I tried running unit tests with `trimpath` it was not showing any details which is weird😞 
[`7929c46` (#120)](https://github.com/gkampitakis/go-snaps/pull/120/commits/7929c467304df6fbfe134686c087aace10266bc5)

This is what ReadBuildInfo doc states
> // ReadBuildInfo returns the build information embedded
> // in the running binary. The information is available only
> // in binaries built with module support.

Reply:
I have updated the function and on my testing 
1. captures all cases, `GOFLAGS`, flag `trimpath` either from a main_test or package
2. don't see any regression on what already exists.